modules/pam_limits/pam_limits.c

```diff
@@ -816,9 +816,22 @@ parse_config_file(pam_handle_t *pamh, const char *uname, uid_t uid, gid_t gid,
         pam_syslog(pamh, LOG_DEBUG, "reading settings from '%s'", CONF_FILE);
     fil = fopen(CONF_FILE, "r");
     if (fil == NULL) {
-        pam_syslog (pamh, LOG_WARNING,
-		    "cannot read settings from %s: %m", CONF_FILE);
-        return PAM_SERVICE_ERR;
+      int err = errno;
+#ifdef VENDORDIR
+
+      /* if the specified file does not exist, and it is not provided by
+	 the user, try the vendor file as fallback. */
```

Comment:
If the idea is to fall back when the file doesn't exist, then `errno` has to be checked to make sure it's `ENOENT` and not something else like `EACCES`.